nextcord/user.py

```diff
@@ -89,13 +95,16 @@ def __hash__(self) -> int:
     def _update(self, data: Union[PartialUserPayload, UserPayload]) -> None:
         self.name = data["username"]
         self.id = int(data["id"])
-        self.discriminator = data["discriminator"]
+        self._discriminator = data.get("discriminator")
         self._avatar = data["avatar"]
         self._banner = data.get("banner", None)
         self._accent_colour = data.get("accent_color", None)
         self._public_flags = data.get("public_flags", 0)
         self.bot = data.get("bot", False)
         self.system = data.get("system", False)
+        self.global_name = data.get("global_name", None)
+
+        self.discriminator = self._discriminator if not self._discriminator == "0" else None
```

Comment:
And same again here (curious as to why you used `not x == y` instead of `x != y` btw).